draft-ietf-quic-v2.md

```diff
@@ -178,23 +178,22 @@ server MUST include the relevant transport parameters to validate that the
 server sent the Retry and the connection IDs used in the exchange, as described
 in {{Section 7.3 of QUIC}}.
 
-The server SHOULD start sending its Initial packets using the negotiated
-version as soon as it decides to change. Before the server is able to process
-transport parameters from the client, it might need to respond to Initial
-packets from the client. For these packets the server uses the original version.
+As the server cannot send CRYPTO frames until it has processed the client's
+transport parameters, it MUST send all CRYPTO frames in Initial packets using
+the negotiated version. The client learns the negotiated version from the first
+Initial CRYPTO frame it successfully processes, and MUST discard any packets
```

Comment:
This describes a protocol violation, so it should close the connection. Though I wouldn't bother checking packet numbers here, it's overhead with limited benefit

Reply:
Making it a protocol violation would allow late-arriving Initial garbage from an attacker to taking down a connection. I've considerably rewritten this now, PTAL.

I've left the door open for a client to accept an OOO v1 ACK, for instance.